docs/concepts/tools/kubectl/object-management-using-declarative-config.md

```diff
@@ -0,0 +1,517 @@
+
+Defaulted fields are *not* re-defaulted unless they are explicitly cleared
+as part of the patch request.
+This may cause unexpected behavior for a fields defaulted to based
+on the values of other fields.  When the other fields are later changed
+the values defaulted from them will not be updated unless they are
+explicitly cleared.
+
+For this reason, it is recommended that certain fields defaulted
+by the server are explicitly defined in the object configuration even
+if the desired values match the server defaults.  This will make it
+easier to recognize conflicting values that will not be re-defaulted
+by the server.
+
+Example:
+
+1. User creates a Deployment without defining `strategy.type`
+2. Server defaults `strategy.type` to `RollingUpdate` and defaults `strategy.rollingUpdate` values
+3. User changes `strategy.type` to `Recreate`.  The `strategy.rollingUpdate` values remain at their defaulted values, though the server expects them to be cleared.
+  - If the strategy.rollingUpdate was defined in the config initially, it would have been more clear that it needed to be deleted.
+4. Apply fails because `strategy.rollingUpdate` is not cleared
+
+Recommendations for fields to explicitly define in the object configuration:
+
+- Selectors and PodTemplate Labels on any controller
+- Deployment RolloutStrategy
+
+### How to clear server defaulted fields or fields set by other writers
+
+As of Kubernetes 1.5, fields that do not appear in the configuration cannot be
+cleared.  For lists that are merged, this means elements of the list
+that do not appear in the configuration cannot be deleted.  The
+following workaround exist:
+
+Option 1: remove the field by directly editing the live object
+
+Option 2: remove the field through the configuration
+- Add the field to the configuration to match the live object
+- Apply the configuration - updates the annotation to include the field
+- Delete the field from the configuration
+- Apply the configuration - delete the field from the live object and annotation.
+
+{% comment %}
+TODO(1.6): Update this with the following for 1.6
+
+Fields that do not appear in the configuration can be cleared by
+setting their values to `null` and then applying the configuration.
+For fields defaulted by the server, this will trigger re-defaulting
+the values.
+{% endcomment %}
+
+## How to change ownership of a field between configuration and live writers
+
+Individual object fields should be owned by a single writer.
+
+### Change owner from a live writer to configuration
+
+Add the field to the configuration.  Stop the live writer from writing to the field.
+
+### Change owner from configuration to a live writer.
+
+As of Kubernetes 1.5, changing ownership of a field from configuration to
+a live writer requires manual steps.
+
+- Remove the field from the configuration
+- Remove the field from the `kubectl.kubernetes.io/last-applied-configuration` annotation on the live object.  This can be done using `edit`
```

Comment:
`edit` may not work for now, since it update `last-applied-config` annotation. Unless we stop `edit` from updating annotation.